### PR TITLE
docs: add explainer on performance budgets

### PR DESCRIPTION
**Pull request by khempenius**
**Related Issues/PRs**
https://github.com/GoogleChrome/lighthouse/issues/10540

### CODE REVIEW

File: docs/performance-budgets.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
+# Performance Budgets (budget.json)
+
+Use a performance budget to prevent site performance from regressing over time.
```
Comment (patrickhulce) [inline docs/performance-budgets.md:3]: in the context of lighthouse we might want a more specific overview explanation, maybe something like

```suggestion
Use a performance budget to assert a custom file size or count limit per resource type and thresholds for each performance metric.
```

? feel free to reword just something unique to budgets would be nice :)

File: docs/performance-budgets.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+Match all URL paths.
+
+`"path": "/"` (This is equivalent to writing `"path": "/*"`)
+
+Match all URL paths starting with `/articles`.
+
+`"path": "/articles"`
+
+Match URL paths within the `uk/` directory and ending with `shopping-cart`.
```
Comment (patrickhulce) [inline docs/performance-budgets.md:150]:
```suggestion
Match URL paths within the `uk/` directory and ending with `/shopping-cart`.
```

right? or maybe I'm misremembering the logic for `/`

aside: would `/uk/shopping-cart` match with this pattern? I *think* the answer is no, so might want to tweak the example so the explanation matches but please ignore if I'm wrong

File: docs/performance-budgets.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+   {
+         "metric": "interactive",
+         "budget": 5000
+   }
+]
+```
+
+Supported timing metrics:
+
+- `first-contentful-paint`
```
Comment (patrickhulce) [inline docs/performance-budgets.md:85]: we should add `largest-contentful-paint` support to this list before 6.0 final :)

**Comment by connorjclark** [comment]
Thanks for putting up this PR so quickly Katie!

File: docs/performance-budgets.md (hunk 1 above)
Comment (connorjclark) [reply docs/performance-budgets.md:3]: related: patrick should we mention interop w/ LHCI here?

File: docs/performance-budgets.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
+# Performance Budgets (budget.json)
+
+Use a performance budget to prevent site performance from regressing over time.
+
+## Usage
+
+Performance budgets are supported in the CLI version of lighthouse. To use:
+1. Create a `budget.json` file.
+2. When running Lighthouse from the command line, pass the `--budget-path` (or `--budgetPath`) flag followed by the path to budget file in order to calculate whenever a category is over budget.
```
Comment (connorjclark) [inline docs/performance-budgets.md:9]:
```suggestion
2. When running Lighthouse from the command line, pass the `--budget-path` flag followed by the path to budget file in order to calculate whenever a category is over budget.
```

don't think that's necessary

File: docs/performance-budgets.md (hunk 3 above)
Comment (connorjclark) [reply docs/performance-budgets.md:85]: Let's just add it now? + CLS. If you don't like that we could at least make an issue now.

File: readme.md [hunk 5]
```diff
@@ -234,6 +234,7 @@ Useful documentation, examples, and recipes to get you started.
 - [Developing Plugins](./docs/plugins.md)
 - [Making a New Audit](./docs/new-audits.md)
 - [Testing on a mobile device](./docs/readme.md#testing-on-a-mobile-device)
+￼- [Setting a performance budget](./docs/performance-budgets.md)
```
Comment (connorjclark) [inline readme.md:237]:
```suggestion
- [Setting a performance budget](./docs/performance-budgets.md)
```

File: docs/performance-budgets.md (hunk 1 above)
Comment (patrickhulce) [reply docs/performance-budgets.md:3]: yeah a good chance to plug the budgets interop there too :)

File: docs/performance-budgets.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
+# Performance Budgets (budget.json)
+
+Use a performance budget to assert thresholds for performance metrics. `budget.json` supports three types of budgets:
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply docs/performance-budgets.md:3]: any opinion here @khempenius or should I feel free to merge? :)

Comment (khempenius) [inline docs/performance-budgets.md:3]:
```suggestion
Use a [performance budget](https://addyosmani.com/blog/performance-budgets/) to assert thresholds for performance metrics. `budget.json` supports three types of budgets:
```

File: docs/performance-budgets.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
+# Performance Budgets (budget.json)
+
+Use a [performance budget](https://addyosmani.com/blog/performance-budgets/) to assert thresholds for performance metrics. `budget.json` supports three types of budgets:
```
Comment (khempenius) [inline docs/performance-budgets.md:3]:
```suggestion
Use a [performance budget](https://web.dev/performance-budgets-101/) to assert thresholds for performance metrics. `budget.json` supports three types of budgets:
```

File: docs/performance-budgets.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (khempenius) [reply docs/performance-budgets.md]: Bah, sorry for the delay - in my notifications it looked like this had been merged before I responded 🤦‍♀️.

I have a deck that I'd like to link here, but it needs to clean up before release - so this link should be good for now.

**Comment by patrickhulce** [comment]
awesome, thanks @khempenius!

**Pull request merged**